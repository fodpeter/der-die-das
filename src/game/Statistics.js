import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import Label from "react-bootstrap/lib/Label";
import "./Statistics.css";
import range from "lodash/range";
import filledHeart from "./heart_filled.svg";
import hollowHeart from "./heart_hollow.svg";

class Statistics extends PureComponent {
  renderHeart = (index, filled) => (
    <img
      key={index}
      className="heart"
      src={filled ? filledHeart : hollowHeart}
      alt={filled ? "Life" : "Empty"}
    />
  );

  renderHearts = () => {
    const { lifes } = this.props;
    return (
      <div>
        Lifes: {range(3).map(index => this.renderHeart(index, index < lifes))}
      </div>
    );
  };

  renderWords = () => {
    const { counter } = this.props;
    return (
      <div className="counter">
        <span>Words:</span>
        <Label bsStyle="success">{counter}</Label>
      </div>
    );
  };

  render = () => (
    <div className="statistics">
      {this.renderHearts()}
      {this.renderWords()}
    </div>
  );
}

Statistics.propTypes = {
  lifes: PropTypes.number.isRequired,
  counter: PropTypes.number.isRequired
};

export default Statistics;
