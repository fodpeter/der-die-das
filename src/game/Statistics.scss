.heart {
  height: 1em;
  margin-right: 0.25em;
  width: 1em;
}

.statistics {
  display: flex;
  justify-content: space-between;
  padding: 0.5em 1em;

  .counter {
    align-items: center;
    display: flex;

    span {
      margin-right: 1ch;
    }
  }
}
