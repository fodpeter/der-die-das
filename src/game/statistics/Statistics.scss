.heart {
  height: 1em;
  margin-right: 0.25em;
  width: 1em;
}

$danger: #d9534f;

.statistics {
  display: flex;
  justify-content: space-between;
  padding: 0.5em 1em;

  .counter {
    align-items: center;
    display: flex;

    span {
      margin-right: 1ch;
    }
  }

  @keyframes text-color-animation {
    25% {
      color: $danger;
    }
    50% {
      color: $danger;
    }
  }

  @keyframes background-color-animation {
    25% {
      background-color: $danger;
    }
    50% {
      background-color: $danger;
    }
  }

  .word-count-enter-active {
    animation-duration: 1000ms;
    animation-name: text-color-animation;

    .label {
      animation-duration: 1000ms;
      animation-name: background-color-animation;
    }
  }
}
