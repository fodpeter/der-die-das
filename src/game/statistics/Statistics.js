import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import Label from "react-bootstrap/lib/Label";
import "./Statistics.css";
import range from "lodash/range";
import filledHeart from "./heart_filled.svg";
import hollowHeart from "./heart_hollow.svg";
import { CSSTransition } from "react-transition-group";

class Statistics extends PureComponent {
  renderHeart = (index, filled) => (
    <img
      key={index}
      className="heart"
      src={filled ? filledHeart : hollowHeart}
      alt={filled ? "Life" : "Empty"}
    />
  );

  renderHearts = () => {
    const { lives } = this.props;
    return (
      <div>
        Lives: {range(3).map(index => this.renderHeart(index, index < lives))}
      </div>
    );
  };

  renderWords = () => {
    const { counter } = this.props;
    return (
      <CSSTransition in={counter === 0} timeout={1000} classNames="word-count">
        <div className="counter">
          <span>Words:</span>
          <Label bsStyle="success">{counter}</Label>
        </div>
      </CSSTransition>
    );
  };

  render = () => (
    <div className="statistics">
      {this.renderWords()}
      {this.renderHearts()}
    </div>
  );
}

Statistics.propTypes = {
  lives: PropTypes.number.isRequired,
  counter: PropTypes.number.isRequired
};

export default Statistics;
