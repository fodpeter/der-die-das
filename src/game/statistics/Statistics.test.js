import React from "react";
import { shallow } from "enzyme";
import Statistics from "./Statistics";

describe("Statistics", () => {
  it("renders without crashing", () => {
    const component = shallow(<Statistics lifes={2} counter={5} />);
    expect(component).toMatchSnapshot();
  });
});
