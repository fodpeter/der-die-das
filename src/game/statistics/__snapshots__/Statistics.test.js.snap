// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Statistics renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Statistics
    counter={5}
    lives={2}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <CSSTransition
          classNames="word-count"
          in={false}
          timeout={1000}
>
          <div
                    className="counter"
          >
                    <span>
                              Words:
                    </span>
                    <Label
                              bsClass="label"
                              bsStyle="success"
                    >
                              5
                    </Label>
          </div>
</CSSTransition>,
        <div>
          Lives: 
          <img
                    alt="Life"
                    className="heart"
                    src="heart_filled.svg"
          />
          <img
                    alt="Life"
                    className="heart"
                    src="heart_filled.svg"
          />
          <img
                    alt="Empty"
                    className="heart"
                    src="heart_hollow.svg"
          />
</div>,
      ],
      "className": "statistics",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <div
            className="counter"
>
            <span>
                        Words:
            </span>
            <Label
                        bsClass="label"
                        bsStyle="success"
            >
                        5
            </Label>
</div>,
          "classNames": "word-count",
          "in": false,
          "timeout": 1000,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                Words:
</span>,
              <Label
                bsClass="label"
                bsStyle="success"
>
                5
</Label>,
            ],
            "className": "counter",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Words:",
              },
              "ref": null,
              "rendered": "Words:",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "label",
                "bsStyle": "success",
                "children": 5,
              },
              "ref": null,
              "rendered": 5,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Lives: ",
            Array [
              <img
                alt="Life"
                className="heart"
                src="heart_filled.svg"
/>,
              <img
                alt="Life"
                className="heart"
                src="heart_filled.svg"
/>,
              <img
                alt="Empty"
                className="heart"
                src="heart_hollow.svg"
/>,
            ],
          ],
        },
        "ref": null,
        "rendered": Array [
          "Lives: ",
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "alt": "Life",
              "className": "heart",
              "src": "heart_filled.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "alt": "Life",
              "className": "heart",
              "src": "heart_filled.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "alt": "Empty",
              "className": "heart",
              "src": "heart_hollow.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <CSSTransition
            classNames="word-count"
            in={false}
            timeout={1000}
>
            <div
                        className="counter"
            >
                        <span>
                                    Words:
                        </span>
                        <Label
                                    bsClass="label"
                                    bsStyle="success"
                        >
                                    5
                        </Label>
            </div>
</CSSTransition>,
          <div>
            Lives: 
            <img
                        alt="Life"
                        className="heart"
                        src="heart_filled.svg"
            />
            <img
                        alt="Life"
                        className="heart"
                        src="heart_filled.svg"
            />
            <img
                        alt="Empty"
                        className="heart"
                        src="heart_hollow.svg"
            />
</div>,
        ],
        "className": "statistics",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <div
              className="counter"
>
              <span>
                            Words:
              </span>
              <Label
                            bsClass="label"
                            bsStyle="success"
              >
                            5
              </Label>
</div>,
            "classNames": "word-count",
            "in": false,
            "timeout": 1000,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span>
                  Words:
</span>,
                <Label
                  bsClass="label"
                  bsStyle="success"
>
                  5
</Label>,
              ],
              "className": "counter",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Words:",
                },
                "ref": null,
                "rendered": "Words:",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "label",
                  "bsStyle": "success",
                  "children": 5,
                },
                "ref": null,
                "rendered": 5,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Lives: ",
              Array [
                <img
                  alt="Life"
                  className="heart"
                  src="heart_filled.svg"
/>,
                <img
                  alt="Life"
                  className="heart"
                  src="heart_filled.svg"
/>,
                <img
                  alt="Empty"
                  className="heart"
                  src="heart_hollow.svg"
/>,
              ],
            ],
          },
          "ref": null,
          "rendered": Array [
            "Lives: ",
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "alt": "Life",
                "className": "heart",
                "src": "heart_filled.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "alt": "Life",
                "className": "heart",
                "src": "heart_filled.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "alt": "Empty",
                "className": "heart",
                "src": "heart_hollow.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
